response/page_path_test.go

```diff
@@ -0,0 +1,356 @@
+				expected: "/economy/inflationandpriceindices/datasets/consumerpriceindicescpiandretailpricesindexrpiitemindicesandpricequotes/pricequotesseptember2023",
+			},
+			{
+				uri:      "/test/image.png",
+				expected: "/test",
+			},
+		}
+		timeSeriesPaths := []TestCase{
+			{
+				uri:      "/economy/grossdomesticproductgdp/timeseries/abmi/pn2/linechartconfig",
+				expected: "/economy/grossdomesticproductgdp/timeseries/abmi/pn2",
+			},
+			{
+				uri:      "/economy/grossdomesticproductgdp/timeseries/abmi/pn2/data/linechartconfig",
+				expected: "/economy/grossdomesticproductgdp/timeseries/abmi/pn2",
+			},
+			{
+				uri:      "/economy/grossdomesticproductgdp/timeseries/abmi/pn2/data/linechartconfig/",
+				expected: "/economy/grossdomesticproductgdp/timeseries/abmi/pn2",
+			},
+			{
+				uri:      "/economy/grossdomesticproductgdp/timeseries/ihyq",
+				expected: "/economy/grossdomesticproductgdp/timeseries/ihyq",
+			},
+			{
+				uri:      "/economy/grossdomesticproductgdp/timeseries/abmi/pn2",
+				expected: "/economy/grossdomesticproductgdp/timeseries/abmi/pn2",
+			},
+			{
+				uri:      "/economy/grossdomesticproductgdp/timeseries/abmi/pn2/data/linechartimage",
+				expected: "/economy/grossdomesticproductgdp/timeseries/abmi/pn2",
+			},
+			{
+				uri:      "/generator?format=csv&uri=/economy/inflationandpriceindices/timeseries/l55o/mm23",
+				expected: "/economy/inflationandpriceindices/timeseries/l55o/mm23",
+			},
+		}
+		datasetsPaths := []TestCase{
+			{
+				uri:      "/economy/inflationandpriceindices/datasets/consumerpriceindicescpiandretailpricesindexrpiitemindicesandpricequotes",
+				expected: "/economy/inflationandpriceindices/datasets/consumerpriceindicescpiandretailpricesindexrpiitemindicesandpricequotes",
+			},
+			{
+				uri:      "/economy/inflationandpriceindices/datasets/consumerpriceindicescpiandretailpricesindexrpiitemindicesandpricequotes/pricequotesseptember2023",
+				expected: "/economy/inflationandpriceindices/datasets/consumerpriceindicescpiandretailpricesindexrpiitemindicesandpricequotes/pricequotesseptember2023",
+			},
+			{
+				uri:      "/economy/inflationandpriceindices/datasets/consumerpriceindices",
+				expected: "/economy/inflationandpriceindices/datasets/consumerpriceindices",
+			},
+			{
+				uri:      "/economy/inflationandpriceindices/datasets/consumerpriceindices/current",
+				expected: "/economy/inflationandpriceindices/datasets/consumerpriceindices/current",
+			},
+			{
+				uri:      "/employmentandlabourmarket/peopleinwork/earningsandworkinghours/datasets/analysisofwageandpriceincreases",
+				expected: "/employmentandlabourmarket/peopleinwork/earningsandworkinghours/datasets/analysisofwageandpriceincreases",
+			},
+			{
+				uri:      "/businessindustryandtrade/changestobusiness/mergersandacquisitions/datasets/timeseries/15march2024",
+				expected: "/businessindustryandtrade/changestobusiness/mergersandacquisitions/datasets/timeseries/15march2024",
```

Comment:
```suggestion
				uri:      "/businessindustryandtrade/changestobusiness/mergersandacquisitions/datasets/timeseries/15march2024/extra-third-segment",
				expected: "/businessindustryandtrade/changestobusiness/mergersandacquisitions/datasets/timeseries/15march2024",
```
should there be a test for the removal of `extra-third-segment`?

Reply:
The URIs you see in this group are real examples. We couldn't find a real example with 3 segments, so I've added a new test case with fake segment names, so that it's clear that even though it's not a real URI, we'd like to test that scenario 👍